core/processor/abci.go

```diff
@@ -1990,14 +2056,32 @@ func (app *App) DeliverStopOrdersSubmission(ctx context.Context, tx abci.Tx, det
 	// Submit the create order request to the execution engine
 	idgen := idgeneration.New(deterministicID)
 	var fallsBelow, risesAbove *string
+	fallsBelowParty := tx.Party()
+	risesAboveParty := tx.Party()
 	if os.FallsBelow != nil {
 		fallsBelow = ptr.From(idgen.NextID())
+		if os.FallsBelow.OrderSubmission != nil {
+			if err := app.verifyVaultOwnership(s.FallsBelow.OrderSubmission.VaultId, tx.Party()); err != nil {
+				return err
+			}
+			if s.FallsBelow.OrderSubmission.VaultId != nil {
+				fallsBelowParty = *s.FallsBelow.OrderSubmission.VaultId
+			}
+		}
 	}
 	if os.RisesAbove != nil {
 		risesAbove = ptr.From(idgen.NextID())
+		if os.RisesAbove.OrderSubmission != nil {
+			if err := app.verifyVaultOwnership(s.RisesAbove.OrderSubmission.VaultId, tx.Party()); err != nil {
+				return err
+			}
+			if s.RisesAbove.OrderSubmission.VaultId != nil {
+				risesAboveParty = *s.RisesAbove.OrderSubmission.VaultId
+			}
+		}
 	}
 
-	_, err = app.exec.SubmitStopOrders(ctx, os, tx.Party(), idgen, fallsBelow, risesAbove)
+	_, err = app.exec.SubmitStopOrders(ctx, os, fallsBelowParty, risesAboveParty, idgen, fallsBelow, risesAbove)
```

Comment:
I realise this is not finished, but here, I'm not sure I understand why we should have two party IDs? I can't see a way for the riseAbove, and fallsBelow party to be ever different?

Reply:
this is finished with respect to what you're asking about. Each of the rises above and falls below is a separate submission, so one can be submitted with a vault id, i.e. on behalf of the vault, and one without so there would be two different parties for falls below and rise above. Whether this should be allow or not is a separate question but it's not defined that it's not allowed as far as I can see.

Reply:
right, I see. I think it should not be allowed, and if submitting those they should both concern the same vault ID, as they can be linked to cancel each other and such.

Reply:
added validation instead that if both are defined they must be equal.